### PR TITLE
Viewer query integration tests

### PR DESCRIPTION
**Pull request by kfurgol**
**Description**

Changes proposed in this pull request:

- Added integration tests for the Viewer query

**Related issue(s)**
#590

### CODE REVIEW

File: tests/end-to-end/e2e/auth_test.go [hunk 1]
```diff
@@ -139,6 +135,22 @@ func fetchHydraAccessToken(t *testing.T, encodedCredentials string, tokenURL str
 	return &hydraToken, nil
 }
 
+func fetchHydraAccessTokenWithScopesForApplication(t *testing.T, encodedCredentials string, tokenURL string) (*hydraToken, error) {
+	form := url.Values{}
+	form.Set("grant_type", "client_credentials")
+	form.Set("scope", "application:write application:read")
```
Comment (kjaksik) [inline tests/end-to-end/e2e/auth_test.go]: I think it would be nice to put the scopes in a const

File: tests/end-to-end/e2e/viewer_test.go [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+		defer unregisterIntegrationSystem(t, ctx, dexGraphQLClient, tenant, intSys.ID)
+
+		t.Log("Generate Client Credentials for Integration System")
+		intSysAuth := generateClientCredentialsForIntegrationSystem(t, ctx, dexGraphQLClient, tenant, intSys.ID)
+		intSysOauthCredentialData, ok := intSysAuth.Auth.Credential.(*graphql.OAuthCredentialData)
+		require.True(t, ok)
+		require.NotEmpty(t, intSysOauthCredentialData.ClientSecret)
+		require.NotEmpty(t, intSysOauthCredentialData.ClientID)
+
+		t.Log("Issue a Hydra token with Client Credentials")
```
Comment (aszecowka) [inline tests/end-to-end/e2e/viewer_test.go]: IMO we should not mention that this is token that comes from Hydra, this is implementation details, so it should be "get access token". Also, those lines:
```
t.Log("Issue a Hydra token with Client Credentials")
oauthCredentials := fmt.Sprintf("%s:%s", intSysOauthCredentialData.ClientID, intSysOauthCredentialData.ClientSecret)
		encodedCredentials := base64.StdEncoding.EncodeToString([]byte(oauthCredentials))
		hydraToken, err := fetchHydraAccessTokenWithScopesForApplication(t, encodedCredentials, intSysOauthCredentialData.URL)
		require.NoError(t, err)
```
can be replaced by calling one function: `getAccessToken`. 

Also, we can get Access token without any scopes, so we can have one func for Apps, Runtimes and Integration Sys

File: tests/end-to-end/e2e/viewer_test.go [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+		oauthGraphQLClient := gql.NewAuthorizedGraphQLClientWithCustomURL(hydraToken.AccessToken, fmt.Sprintf("https://compass-gateway-auth-oauth.%s/director/graphql", domain))
+
+		t.Log("Requesting Viewer as Integration System")
+		viewer := graphql.Viewer{}
+		req := fixGetViewerRequest()
+
+		err = tc.RunOperationWithCustomTenant(ctx, oauthGraphQLClient, tenant, req, &viewer)
+		require.NoError(t, err)
+		assert.Equal(t, intSys.ID, viewer.ID)
+		assert.Equal(t, graphql.ViewerTypeIntegrationSystem, viewer.Type)
```
[the remark this answers is not in the document]
Comment (kfurgol) [reply tests/end-to-end/e2e/viewer_test.go]: Is it supposed to be deleted explicitly? Right now `system auth` is being deleted with deletion of Integration System.

[the remark this answers is not in the document]
Comment (aszecowka) [reply tests/end-to-end/e2e/viewer_test.go]: so it is not needed

File: tests/end-to-end/e2e/README.md [hunk 4]
```diff
@@ -1,9 +1,9 @@
 # Compass end to end tests
 
 ## Test Scenario:
```
Comment (klaudiagrz) [inline tests/end-to-end/e2e/README.md]:
```suggestion
Compass end to end test scenario consists of the following steps:
```

File: tests/end-to-end/e2e/README.md [hunk 5]
```diff
@@ -1,9 +1,9 @@
...
-- Get Dex token, create IntegrationSystem with it and generate client credentials for IntSystem
+- Get Dex token, create IntegrationSystem with it and request client credentials for IntSystem
```
Comment (klaudiagrz) [inline tests/end-to-end/e2e/README.md]:
```suggestion
- Get a Dex token, create the Integration System using this token, and request client credentials for the IntegrationSystem
```

File: tests/end-to-end/e2e/README.md [hunk 6]
```diff
@@ -1,9 +1,9 @@
...
 - Call Hydra for OAuth 2.0 access token with client_id and client_secret pair - https://github.com/kyma-incubator/examples/tree/master/ory-hydra/scenarios/client-credentials
-- Create an Application (as Integration System)
+- Register an Application (as Integration System)
```
Comment (klaudiagrz) [inline tests/end-to-end/e2e/README.md]:
```suggestion
- Register an Application as the Integration System
```

File: tests/end-to-end/e2e/README.md [hunk 7]
```diff
@@ -21,5 +21,7 @@ To run the test locally, set these environment variables:
 | USER_PASSWORD |    Dex static user password   |   - |
 | DEFAULT_TENANT | Default tenant value |    `3e64ebae-38b5-46a0-b1ed-9ccee153a0ae` |
 | DOMAIN | Kyma domain name |    `kyma.local` |
-
+| GATEWAY_JWTSUBDOMAIN | Default gateway for handling requests with JWT | compass-gateway |
+| GATEWAY_CLIENT_CERTS_SUBDOMAIN | Default gateway for handling requests with certificate | compass-gateway-mtls |
+| GATEWAY_OAUTH20_SUBDOMAIN | Default gateway forhandling requests with OAuth access token | compass-gateway-auth-oauth|
```
Comment (klaudiagrz) [inline tests/end-to-end/e2e/README.md]:
```suggestion
| GATEWAY_OAUTH20_SUBDOMAIN | Default gateway for handling requests with an OAuth access token | compass-gateway-auth-oauth|
```

File: tests/end-to-end/e2e/README.md [hunk 8]
```diff
@@ -21,5 +21,7 @@ To run the test locally, set these environment variables:
 | USER_PASSWORD |    Dex static user password   |   - |
 | DEFAULT_TENANT | Default tenant value |    `3e64ebae-38b5-46a0-b1ed-9ccee153a0ae` |
 | DOMAIN | Kyma domain name |    `kyma.local` |
-
+| GATEWAY_JWTSUBDOMAIN | Default gateway for handling requests with JWT | compass-gateway |
+| GATEWAY_CLIENT_CERTS_SUBDOMAIN | Default gateway for handling requests with certificate | compass-gateway-mtls |
```
Comment (klaudiagrz) [inline tests/end-to-end/e2e/README.md]:
```suggestion
| GATEWAY_CLIENT_CERTS_SUBDOMAIN | Default gateway for handling requests with a certificate | compass-gateway-mtls |
```

File: tests/end-to-end/e2e/README.md [hunk 9]
```diff
@@ -21,5 +21,7 @@ To run the test locally, set these environment variables:
 | USER_PASSWORD |    Dex static user password   |   - |
 | DEFAULT_TENANT | Default tenant value |    `3e64ebae-38b5-46a0-b1ed-9ccee153a0ae` |
 | DOMAIN | Kyma domain name |    `kyma.local` |
-
+| GATEWAY_JWTSUBDOMAIN | Default gateway for handling requests with JWT | compass-gateway |
```
Comment (klaudiagrz) [inline tests/end-to-end/e2e/README.md]:
```suggestion
| GATEWAY_JWTSUBDOMAIN | Default gateway for handling requests with a JWT | compass-gateway |
```

**Pull request merged**